### PR TITLE
sig-windows: add --docker-config-file argument

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: Ernest Wong <chuwon@microsoft.com>

Related PR: https://github.com/kubernetes-sigs/windows-testing/pull/259

- we need to use e2eprivate dockerhub instead of the default `gcAuthenticatedRegistry` because it does not support 2004 and 20H2
- removed preset-windows-repo-list-2004 since it's no longer needed
- changed path alias from k8s.io/windows-testing to sigs.k8s.io/windows-testing

/assign @jsturtevant 
/hold

### CODE REVIEW

**Comment by jsturtevant** [comment]
are there any other gcr tests that need these setting?

File: config/jobs/kubernetes/sig-windows/windows-gce.yaml [hunk 1]
```diff
@@ -83,6 +83,7 @@ periodics:
       - --test_args=--ginkgo.focus=\[Conformance\]|\[NodeConformance\]|\[sig-windows\] --ginkgo.skip=\[LinuxOnly\]|\[Serial\]|\[Feature:.+\] --minStartupPods=8
       - --test-cmd-args=--ginkgo.focus=\[Conformance\]|\[NodeConformance\]|\[sig-windows\] --ginkgo.skip=\[LinuxOnly\]|\[Serial\]|\[Feature:.+\] --minStartupPods=8
       - --test-cmd-args=--node-os-distro=windows
+      - --test-cmd-args=--docker-config-file=$(DOCKER_CONFIG_FILE)
```
Comment (jsturtevant) [inline config/jobs/kubernetes/sig-windows/windows-gce.yaml:86]: I think we are missing `preset-windows-private-registry-cred: "true"` on this job?

File: config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml [hunk 2]
```diff
@@ -897,7 +902,6 @@ periodics:
     preset-service-account: "true"
     preset-azure-cred: "true"
     preset-azure-windows: "true"
-    preset-windows-repo-list-master: "true"
```
Comment (jsturtevant) [inline config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml:900]: the repolists are not needed for the azure disk/file prs?

Comment (chewong) [reply config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml:900]: No, they do not use it at all. Just piggy riding this PR to clean them up.

**Comment by jsturtevant** [comment]
@jeremyje could you confirm we got the config right for the gce jobs?

**Comment by jsturtevant** [comment]
lgtm but will wait for @jeremyje to give the ok on the gce jobs

**Comment by jeremyje** [comment]
https://github.com/kubernetes/test-infra/blob/4dd1251b5618b1a813d455f563d059fb6ee2dde4/config/jobs/kubernetes/sig-windows/windows-gce.yaml is where the GCE jobs be at.

**Comment by chewong** [comment]
Rebased. PTAL

**Review by marosset — APPROVED** [review]
/lgtm

**Pull request merged**